sdk/identity/identity/test/public/node/azurePipelinesCredential.spec.ts

```diff
@@ -0,0 +1,96 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT license.
+
+import { AzurePipelinesCredential } from "../../../src";
+import { isLiveMode } from "@azure-tools/test-recorder";
+import { assert } from "@azure-tools/test-utils";
+
+describe("AzurePipelinesCredential", function () {
+  const scope = "https://vault.azure.net/.default";
+  const tenantId = process.env.AZURE_SERVICE_CONNECTION_TENANT_ID!;
+
+  it("authenticates with a valid service connection", async function () {
+    if (!isLiveMode() || !process.env.AZURE_SERVICE_CONNECTION_ID) {
+      this.skip();
+    }
+    // this serviceConnection corresponds to the Azure SDK Test Resources - LiveTestSecrets service
+    const existingServiceConnectionId = process.env.AZURE_SERVICE_CONNECTION_ID!;
+    // clientId for above service connection
+    const clientId = process.env.AZURE_SERVICE_CONNECTION_CLIENT_ID!;
+    const systemAccessToken = process.env.SYSTEM_ACCESSTOKEN!;
+    const credential = new AzurePipelinesCredential(
+      tenantId,
+      clientId,
+      existingServiceConnectionId,
+      systemAccessToken
+    );
+    try {
+      const token = await credential.getToken(scope);
+      assert.ok(token?.token);
```

Comment:
Check out [assert.isNotEmpty](https://www.chaijs.com/api/assert/#method_isnotempty) for this assertion which is better than the old `assert.ok` check